src/intra_edge.rs

```diff
@@ -214,34 +268,18 @@ impl<const SB128: bool, const N_BRANCH: usize, const N_TIP: usize>
         top_has_right: bool,
         left_has_bottom: bool,
     ) -> (Self, EdgeIndices) {
-        let mut branch = EdgeBranch::new(
-            (if top_has_right {
-                EDGE_TOP_HAS_RIGHT
-            } else {
-                0 as EdgeFlags
-            }) | (if left_has_bottom {
-                EDGE_LEFT_HAS_BOTTOM
-            } else {
-                0 as EdgeFlags
-            }),
-            bl,
-        );
+        let thr = EdgeFlags::TOP_HAS_RIGHT.select(top_has_right);
+        let lhb = EdgeFlags::LEFT_HAS_BOTTOM.select(left_has_bottom);
+        let mut branch = EdgeBranch::new(thr.union(lhb), bl);
```

Comment:
I think I prefer an inline `union_all` since `thr` and `lhb` are kinda confusing.